src/dynamic-checkout/clients/apple-pay.ts

```diff
@@ -36,22 +61,48 @@ module ProcessOut {
     ) {
       const tokenizeApplePay = this.tokenizeApplePay.bind(this);
 
-      this.processOutInstance.applePay.checkAvailability(function (err) {
-        if (err) {
-          console.log(err);
-        } else {
-          buttonContainer.classList.add("visible");
-          buttonContainer.addEventListener("click", () => {
-            tokenizeApplePay(invoiceData, getViewContainer);
-          });
+      const applePayPaymentMethodData =
+        this.getApplePayPaymentMethodData(invoiceData);
+
+      this.processOutInstance.applePay.checkAvailability(
+        function (err) {
+          if (err) {
+            console.log(err);
+          } else {
+            buttonContainer.classList.add("visible");
+
+            document
+              .querySelector("apple-pay-button")
+              .addEventListener("click", () => {
+                tokenizeApplePay(invoiceData, getViewContainer);
+              });
+          }
+        },
+        { merchantApplePayCertificateId: applePayPaymentMethodData.merchant_id }
+      );
+    }
+
+    private getSupportedNetworks(invoiceData: Invoice) {
+      const applePayPaymentMethodData =
+        this.getApplePayPaymentMethodData(invoiceData);
+
+      let supporedNetworks = [];
+
+      applePayPaymentMethodData.supported_networks.forEach((network) => {
+        if (this.networksMap[network]) {
+          supporedNetworks.push(this.networksMap[network]);
```

Comment:
Typo in `supporedNetworks` but I also wonder if filter could be used here instead?

Reply:
.filter doesn't work in PO.js since it was introduced in newer versions of JS 😄 that's why every time we need to filter something we need to use forEach. I will fix the typo, thanks!